src/axom/inlet/tests/inlet_function.cpp

```diff
@@ -0,0 +1,269 @@
+// Copyright (c) 2017-2020, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+#include "gtest/gtest.h"
+
+#include <array>
+#include <string>
+#include <vector>
+#include <unordered_map>
+
+#include <iostream>
+
+#include "axom/sidre.hpp"
+
+#include "axom/inlet/LuaReader.hpp"
+#include "axom/inlet/Inlet.hpp"
+
+using axom::inlet::FunctionType;
+using axom::inlet::Inlet;
+using axom::inlet::InletType;
+using axom::inlet::LuaReader;
+using axom::sidre::DataStore;
+
+Inlet createBasicInlet(DataStore* ds,
+                       const std::string& luaString,
+                       bool enableDocs = true)
+{
+  auto lr = std::make_unique<LuaReader>();
+  lr->parseString(luaString);
+
+  return Inlet(std::move(lr), ds->getRoot(), enableDocs);
+}
+
+TEST(inlet_function, simple_vec3_to_double_raw)
+{
+  std::string testString = "function foo (x, y, z) return x + y + z end";
+  DataStore ds;
+  auto inlet = createBasicInlet(&ds, testString);
+
+  auto func =
+    inlet.reader().getFunction("foo", FunctionType::Double, {FunctionType::Vec3D});
+
+  EXPECT_TRUE(func);
+  auto result = func.call<double>(axom::primal::Vector3D {1, 2, 3});
+  EXPECT_FLOAT_EQ(result, 6);
+}
+
+TEST(inlet_function, simple_vec3_to_vec3_raw)
+{
+  std::string testString = "function foo (x, y, z) return 2*x, 2*y, 2*z end";
+  DataStore ds;
+  auto inlet = createBasicInlet(&ds, testString);
+
+  auto func =
+    inlet.reader().getFunction("foo", FunctionType::Vec3D, {FunctionType::Vec3D});
+
+  EXPECT_TRUE(func);
+  auto result =
+    func.call<axom::primal::Vector3D>(axom::primal::Vector3D {1, 2, 3});
```

Comment:
Hmm, I currently have `inlet::FunctionType` as an enum whose values are integers instead of types (to avoid introducing templates).  One option might be to define a trait, where `inlet::AsType<FunctionType::Vec3D>::type` would be an alias for `primal::Vector3D`, but that might be a little verbose.

Reply:
My comment was more of an observation than an actionable item, but this could be somewhat confusing.

Reply:
I think your suggestion to use Inlet aliases for these "external" (to Inlet) types could help with this.
A trait (either enum -> type or type -> enum) could also help improve consistency.